ballerina-tests/tests/interceptors.bal

```diff
@@ -0,0 +1,284 @@
+    *graphql:Interceptor;
+
+    isolated remote function execute(graphql:Context ctx, graphql:Field 'field) returns anydata|error {
+        var result = ctx.resolve();
+        if result is record{} {
+            return {
+                name: "Albus Percival Wulfric Brian Dumbledore"
+            };
+        }
+        return result;
+    }
+}
+
+readonly service class InvalidInterceptor1 {
+    *graphql:Interceptor;
+
+    isolated remote function execute(graphql:Context ctx, graphql:Field 'field) returns anydata|error {
+        var result = ctx.resolve();
+        if result is graphql:ErrorDetail {
+            return {
+                name: "Albus Percival Wulfric Brian Dumbledore",
+                age: 80,
+                address: {number: "103", street: "Mould-on-the-Wold", city: "London"}
+            };
+        }
+        return result;
+    }
+}
+
+readonly service class InvalidInterceptor2 {
+    *graphql:Interceptor;
+
+    isolated remote function execute(graphql:Context ctx, graphql:Field 'field) returns anydata|error {
+        var result = ctx.resolve();
+        if result is int {
+            return ["Ballerina", "GraphQL"];
+        }
+        return result;
+    }
+}
+
+readonly service class InvalidInterceptor3 {
+    *graphql:Interceptor;
+
+    isolated remote function execute(graphql:Context ctx, graphql:Field 'field) returns anydata|error {
+        var result = ctx.resolve();
+        if result is record{} {
+            return "Harry Potter";
+        }
+        return result;
+    }
+}
+
+readonly service class InvalidInterceptor4 {
+    *graphql:Interceptor;
+
+    isolated remote function execute(graphql:Context ctx, graphql:Field 'field) returns anydata|error {
+        var result = ctx.resolve();
+        if result is anydata[] {
+            return {
+                name: "Albus Percival Wulfric Brian Dumbledore",
+                age: 80,
+                address: {number: "103", street: "Mould-on-the-Wold", city: "London"}
+            };
+        }
+        return result;
+    }
+}
+
+readonly service class InvalidInterceptor5 {
+    *graphql:Interceptor;
+
+    isolated remote function execute(graphql:Context ctx, graphql:Field 'field) returns anydata|error {
+        var result = ctx.resolve();
+        if result is record{} {
+            return "Harry Potter";
+        }
+        return result;
+    }
+}
+
+readonly service class InvalidInterceptor6 {
+    *graphql:Interceptor;
+
+    isolated remote function execute(graphql:Context ctx, graphql:Field 'field) returns anydata|error {
+        var result = ctx.resolve();
+        if result is record{} {
+            return ["Ballerina", "GraphQL"];
+        }
+        return result;
+    }
+}
+
+readonly service class InvalidInterceptor7 {
+    *graphql:Interceptor;
+
+    isolated remote function execute(graphql:Context ctx, graphql:Field 'field) returns anydata|error {
+        var result = ctx.resolve();
```

Comment:
Every interceptor here calls the `ctx.resolve()` in the first line and then processes it. We should have a test to execute something before, then call `resolve` and then do another thing. We need to test this functionality whether it executes in the same order as intended.

Reply:
Added a test.